### PR TITLE
Update docstrings in mlflow/tracing

### LINKED ISSUE #12082: Fix docstrings in `mlflow/tracing`
The following file contains incorrect docstrings. We want to fix them.

```
mlflow/tracing/export/inference_table.py:52:9: D417 Missing argument description in the docstring for `export`: `root_spans`
mlflow/tracing/export/mlflow.py:44:9: D417 Missing argument description in the docstring for `export`: `root_spans`
mlflow/tracing/provider.py:35:5: D417 Missing argument description in the docstring for `start_detached_span`: `experiment_id`
mlflow/tracing/utils.py:122:5: D417 Missing argument description in the docstring for `deduplicate_span_names_in_place`: `spans`
```

### PR DESCRIPTION
**Pull request by harupy**
This PR is an attempt to fix minor issues using https://githubnext.com/projects/copilot-workspace/

<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/12083?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12083/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12083
```

</p>
</details>

Resolve #12082

Updates docstrings in the `mlflow/tracing` module to include missing argument descriptions and correct inaccuracies.

- Adds a description for the `root_spans` argument in the `export` method docstrings of both `mlflow/tracing/export/inference_table.py` and `mlflow/tracing/export/mlflow.py`, clarifying that only root spans are passed to these methods.
- Enhances the `start_detached_span` method docstring in `mlflow/tracing/provider.py` to include a description for the `experiment_id` argument, specifying its use for associating the span with a specific experiment in MLflow.
- Modifies the `deduplicate_span_names_in_place` method docstring in `mlflow/tracing/utils.py` to accurately describe the `spans` argument, detailing the deduplication process applied to span names within a list.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/mlflow/mlflow/issues/12082?shareId=8da3d055-a311-4cc0-a84c-4cd5c92e1f87).

---

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by harupy** [comment]
@mlflow-automation autoformat

File: mlflow/tracing/export/inference_table.py
Comment (harupy) [inline mlflow/tracing/export/inference_table.py:1]: Needed a few manual changes but copilot almost nailed it.

File: mlflow/tracing/export/mlflow.py [hunk 1]
```diff
@@ -46,8 +46,9 @@ def export(self, root_spans: Sequence[ReadableSpan]):
         Export the spans to MLflow backend.
 
         Args:
-            spans: A sequence of OpenTelemetry ReadableSpan objects to be exported.
-                Only root spans for each trace are passed to this method.
+            root_spans: A sequence of OpenTelemetry ReadableSpan objects to be exported.
+                Only root spans for each trace are passed to this method. The root spans are
+                the spans that have no parent span.
```
Comment (harupy) [inline mlflow/tracing/export/mlflow.py:51]: Copilot added `The root spans are the spans that have no parent span.`

File: mlflow/tracing/provider.py [hunk 2]
```diff
@@ -43,6 +43,8 @@ def start_detached_span(
         name: The name of the span.
         parent: The parent OpenTelemetry span. If not provided, the span will be created as a root
                 span.
+        experiment_id: The ID of the experiment. This is used to associate the span with a specific
+                       experiment in MLflow.
```
Comment (harupy) [inline mlflow/tracing/provider.py:47]:
```suggestion
        experiment_id: The ID of the experiment. This is used to associate the span with a specific
            experiment in MLflow.
```

File: mlflow/tracing/export/mlflow.py (hunk 1 above)
Comment (harupy) [reply mlflow/tracing/export/mlflow.py:51]:
```suggestion
            root_spans: A sequence of OpenTelemetry ReadableSpan objects to be exported.
                Only root spans for each trace are passed to this method.
```

File: mlflow/tracing/export/inference_table.py [hunk 3]
```diff
@@ -54,8 +54,9 @@ def export(self, root_spans: Sequence[ReadableSpan]):
         Export the spans to MLflow backend.
 
         Args:
-            spans: A sequence of OpenTelemetry ReadableSpan objects to be exported.
-                Only root spans for each trace are passed to this method.
+            root_spans: A sequence of OpenTelemetry ReadableSpan objects to be exported.
+                Only root spans for each trace are passed to this method. The root spans are
+                the spans that have no parent span.
```
Comment (harupy) [inline mlflow/tracing/export/inference_table.py:59]:
```suggestion
            root_spans: A sequence of OpenTelemetry ReadableSpan objects to be exported.
                Only root spans for each trace are passed to this method.
```

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Pull request merged**